src/python/pants/backend/python/util_rules/pex_test.py

```diff
@@ -654,3 +681,179 @@ def _run_pex_for_lockfile_test(
             f"--python-setup-invalid-lockfile-behavior={behavior}",
         ),
     )
+
+
+@pytest.mark.parametrize(
+    "lockfile_type,invalid_reqs,invalid_constraints,uses_source_plugins,uses_project_ic",
+    [
+        (lft, ir, ic, usp, upi)
+        for lft in LOCKFILE_TYPES
+        for ir in BOOLEANS
+        for ic in BOOLEANS
+        for usp in BOOLEANS
+        for upi in BOOLEANS
+        if (ir or ic)
+    ],
+)
+def test_validate_metadata(
```

Comment:
You're right - this does make a already big file even bigger and more complex. And this will only get worse when we add multiple user lockfiles.

What do you think about this home being in `lockfile_metadata.py`? Maybe it'd even make sense to rename to `lockfile_invalidation.py`?

Reply:
Sounds like a good fix for the next round of changes :)

Reply:
Sgtm!